src/modules/rppt_tensor_effects_augmentations.cpp

```diff
@@ -1452,4 +1602,199 @@ RppStatus rppt_vignette_gpu(RppPtr_t srcPtr,
 #endif // backend
 }
 
+/******************** pixelate ********************/
+
+RppStatus rppt_pixelate_gpu(RppPtr_t srcPtr,
+                            RpptDescPtr srcDescPtr,
+                            RppPtr_t dstPtr,
+                            RpptDescPtr dstDescPtr,
+                            Rpp32f pixelationPercentage,
+                            RpptROIPtr roiTensorPtrSrc,
+                            RpptRoiType roiType,
+                            rppHandle_t rppHandle)
+{
+#ifdef HIP_COMPILE
+
+    RpptInterpolationType interpolationType = RpptInterpolationType::BILINEAR;
+    RpptDesc interDesc;
+    interDesc = *srcDescPtr;
+    RpptDescPtr interDescPtr = &interDesc;
+    RpptImagePatchPtr internalDstImgSizes;
+    hipHostMalloc(&internalDstImgSizes, sizeof(RpptImagePatch) * rpp::deref(rppHandle).GetBatchSize());
+    RpptROIPtr internalRoiTensorPtrSrc;
+    hipHostMalloc(&internalRoiTensorPtrSrc, sizeof(RpptROI) * rpp::deref(rppHandle).GetBatchSize());
```

Comment:
Similar comment as Host. Pls use reinterpret cast and scratchBufferHip for these.